compiler/noirc_errors/src/position.rs

```diff
@@ -109,12 +109,17 @@ impl From<Range<u32>> for Span {
     }
 }
 
-impl chumsky::Span for Span {
+impl chumsky::span::Span for Span {
     type Context = ();
 
     type Offset = u32;
 
-    fn new(_context: Self::Context, range: Range<Self::Offset>) -> Self {
+    fn new(_context: Self::Context, mut range: Range<Self::Offset>) -> Self {
+        // TODO why chumsky reports reversed ranges??
+        if range.start > range.end {
+            range = Range { start: range.end, end: range.start };
+        } else {
+        }
```

Comment:
This has been a problem in the past when using certain error recovery strategies I think. Hopefully it hasn't gotten more common with 1.0 😅